Vineyard/Vineyard/GroupListView.swift

```diff
@@ -0,0 +1,74 @@
+//
+//  GroupListView.swift
+//  Vineyard
+//
+//  Created by Jiyoon Lee on 10/20/24.
+//
+
+import SwiftUI
+
+struct GroupListView: View {
+    @StateObject var viewModel = GroupsListViewModel()
+    @State private var isPresentingAddGroup = false
+    var body: some View {
+        NavigationView {
+            ScrollView {
+                VStack(alignment: .center, spacing: 30) {
+                    Spacer()
+                    HStack {
+                        VStack(alignment: .leading) {
+                            Text("Your Groups")
+                                .font(.system(size: 30, weight: .bold))
+                                .foregroundColor(.white)
+                            Text("\(viewModel.user.groups.count) active group(s)")
+                                .foregroundColor(.white)
+                            
+                        }
+                        Spacer()
+                        Button(action: {
+                            print("changing mode")
+                        }) {
+                            Image(systemName: "list.bullet")
+                                .foregroundColor(.white)
+                                .font(.system(size: 25))
+                        }
+                        Button(action: {
+                            print("add")
+                            isPresentingAddGroup = true
+                        }) {
+                            Image(systemName: "plus")
+                                .foregroundColor(.white)
+                                .font(.system(size: 25))
+                        }
+                    }
+                    .padding(40)
+                    
+                    Spacer()
+                    VStack(spacing: 20) {
+                        ForEach(viewModel.user.groups, id: \.self) { groupId in
+                            if let group = viewModel.group(id: groupId) {
+                                NavigationLink(destination: GroupView(group: group)) {
+                                    GroupCardView(group: group).padding()
+                                }
+                                .buttonStyle(PlainButtonStyle())
+                            } else {
+                                Text("Group not found")
+                            }
+                        }.padding()
+                    }
+                }
+                .navigationBarHidden(true)
+                .background(alignment: .top) {
+                    Image("topBackground")
+                }
+            }.ignoresSafeArea()
+        }
+        .sheet(isPresented: $isPresentingAddGroup) {
+            AddGroupView(isPresented: $isPresentingAddGroup, viewModel: viewModel)
```

Comment:
Does this use Yash's new pipeline implemented in #41?

Reply:
ye
![Simulator Screenshot - iPhone 16 Pro - 2024-11-03 at 12 59 17](https://github.com/user-attachments/assets/aeaa961b-5c90-49e4-a6d9-490c107ee27f)

Reply:
No, there's a new pipeline that Yash has. Pull from main and see

Reply:
sorry I saw the video and didn't realize it is differentt
I changed so that it is using CreateGroupView rather than AddGroupView

https://github.com/user-attachments/assets/e1aa9765-ab73-4fa6-a84c-2e66da9d89f5